app/controllers/admin/discourse_calendar/admin_holidays_controller.rb

```diff
@@ -1,19 +1,39 @@
 # frozen_string_literal: true
 
-require "holidays"
-
 module Admin::DiscourseCalendar
   class AdminHolidaysController < AdminDiscourseCalendarController
     def index
       region_code = params[:region_code]
 
       begin
-        holidays = Holidays.year_holidays([region_code], Time.current.beginning_of_year)
+        holidays = DiscourseCalendar::Holiday.find_holidays_for(region_code: region_code)
       rescue Holidays::InvalidRegion
         return render_json_error(I18n.t("system_messages.discourse_calendar_holiday_region_invalid"), 422)
       end
 
       render json: { region_code: region_code, holidays: holidays }
     end
+
+    def disable
+      DiscourseCalendar::DisabledHoliday.create(disabled_holiday_params)
+      CalendarEvent.delete_by(
+        description: disabled_holiday_params[:holiday_name],
+        region: disabled_holiday_params[:region_code]
+      )
+
+      render json: success_json
+    end
+
+    def enable
+      DiscourseCalendar::DisabledHoliday.delete_by(disabled_holiday_params)
```

Comment:
```suggestion
      DiscourseCalendar::DisabledHoliday.destroy_by(disabled_holiday_params)
```

Should we also handle the case where there's no record?

Reply:
I updated this to use `destroy_by`. I added a check to see if a record was deleted, if one was deleted it'll return a 200 and if one wasn't it'll return a 422. I added some tests for these cases too.